source/digits_hits/include/GateJPETGammaPolarizationActor.hh

```diff
@@ -0,0 +1,302 @@
+	 * This function used by GateVActor::BeginOfRunAction. If you want to reset data in file for each new run, call EnableResetDataAtEachRun(true).
+	 * By default this function is not used;
+	 */
+	virtual void ResetData();
+
+	/**Set log file
+	 * @param: log_file_name - log file name
+	 * */
+	void SetLogFile(G4String log_file_name);
+
+	/** Set saving numerical data to ASCI file
+	 * @param: diag_file_name - ASCI file name
+	 * */
+	void SetDiagnosticFile(G4String diag_file_name);
+
+	/** Set filtration range for phi
+	 * @param: value - value to set - depends on what is mode you have to input other value (for more look to function body)
+	 * @param: mode - mode inform how to interpret value
+	*/
+	void SetPhiFlagParameter(G4double value, GateJPETGammaPolarizationActor::PhiThetaValueMode mode);
+
+	/** Set filtration range for theta
+	 * @param: value - value to set - depends on what is mode you have to input other value (for more look to function body)
+	 * @param: mode - mode inform how to interpret value
+	*/
+	void SetThetaFlagParameter(G4double value, GateJPETGammaPolarizationActor::PhiThetaValueMode mode);
+
+	/** Set filtration range for phi
+	 * @param: setAsPlus - if is necessary to save phi angle as only positive value this flag must be set as true
+	*/
+	void SetAllPhiAsNoNegative(G4bool setAsPlus);
+
+	/** Set generation of histogram as (dis)enable.
+	 * By default generation is enable;
+	 * @param: enable - set false to disable histograms generations
+	*/
+	void SetHistogramsEnable(G4bool enable);
+
+	/** Set generation of tests branch (for polarization) as (dis)enable.
+	 * By default generation is enable;
+	 * @param: enable - set false to disable tests branch generations
+	*/
+	void SetTestsBranchEnable(G4bool enable);
+
+	/** Set angle precision per bin in histograms - this mean that each bin has width equal angle_precison
+	 * By default precision is 2 degree.
+	 * @param: angle_precision - bin width (in degrees)
+	 * @exceptions: If value is to huge, what mean angle_precision >= 180.0 deg then this value is ignored.
+	*/
+	void SetAnglePrecisionPerBinInHistograms(G4double angle_precision);
+
+	/** Set (dis)enable saving data ony for desired number of particles per event which scattered.
+	 * The purpose of this to reduce number of data generated during simulation.
+	 * @param: enable - set true to use this, false - otherwise
+	 * */
+	void SetSaveOnlyWhenTheDesiredNumberOfParticlesHasScatteredFromEventEnable(G4bool enable);
+
+	/** Set desired number of particles scattered per event.
+	 * @param: particles_number - if number of scattered per event is equal particles_number then data will be saved
+	 * */
+	void SetDesiredNumberOfParticlesScatteredPerEvent(G4int particles_number);
+
+private: //Functions
+	/** Function generate perpendicular vector to vector 'a'
+	 * @param: a - vector for which we create perpendicular one
+	 * @return: perpendicular vector to 'a'
+	*/
+	G4ThreeVector SetPerpendicularVector(G4ThreeVector& a);
```

Comment:
so the argument should be const G4ThreeVector& a

Reply:
yup